hail/hail/src/is/hail/sparkextras/OriginUnionRDD.scala

```diff
@@ -31,13 +34,13 @@ class OriginUnionRDD[T: ClassTag, S: ClassTag](
   }
 
   override def getDependencies: Seq[Dependency[_]] = {
-    val deps = new ArrayBuffer[Dependency[_]]
+    val deps = ArraySeq.newBuilder[Dependency[_]]
     var i = 0
     for (rdd <- rdds) {
       deps += new RangeDependency(rdd, 0, i, rdd.partitions.length)
       i += rdd.partitions.length
     }
-    deps
+    deps.result()
```

Comment:
Doesnt result copy the resulting array? Do we need to do that if we know the size of the array beforehand?

Reply:
See my other comment. Though it looks like we actually can give a size hint here, and so avoid copying.